prov/tcp/src/tcpx.h

```diff
@@ -125,18 +125,34 @@ enum tcpx_cm_state {
 	TCPX_EP_ERROR,
 };
 
-struct tcpx_msg_hdr {
-	struct ofi_op_hdr	hdr;
+struct tcpx_base_hdr {
+	uint8_t			version;
+	uint8_t			op;
+	uint16_t		flags;
+	uint8_t			op_data;
+	uint8_t			rsvd;
+	uint16_t		payload_off;
+	uint64_t		size;
+};
+
+struct tcpx_rma_hdr {
 	size_t			rma_iov_cnt;
-	union {
-		struct fi_rma_iov	rma_iov[TCPX_IOV_LIMIT];
-		struct fi_rma_ioc	rma_ioc[TCPX_IOV_LIMIT];
-	};
+	struct fi_rma_iov	rma_iov[TCPX_IOV_LIMIT];
```

Comment:
> rma_iov should probably be an empty array [], with the size determined by the iov_cnt.

Since we want to have inject buffer as well rma_iov in the header memory chunk. Both can't be of variable length while having a header structures.  That's why I am using offsets instead of different header definitions.